### PR TITLE
udev: net_id: add support for phys_port_name attribute

### PR DESCRIPTION
**Pull request by jpirko**
Switch drivers uses phys_port_name attribute to pass front panel port
name to user. Use it to generate netdev names.

Signed-off-by: Jiri Pirko jiri@mellanox.com

### CODE REVIEW

File: src/udev/udev-builtin-net_id.c [hunk 1]
```diff
@@ -164,11 +166,16 @@ static int dev_pci_onboard(struct udev_device *dev, struct netnames *names) {
...
+        /* kernel provided front panel port name for multiple port PCI device */
+        port_name = udev_device_get_sysattr_value(dev, "phys_port_name");
+
         s = names->pci_onboard;
         l = sizeof(names->pci_onboard);
         l = strpcpyf(&s, l, "o%d", idx);
         if (dev_port > 0)
                 l = strpcpyf(&s, l, "d%d", dev_port);
+        if (port_name)
+                l = strpcpyf(&s, l, "n%s", port_name);
```
Comment (keszybz) [inline src/udev/udev-builtin-net_id.c:176]: Does it make sense to have both port_name and dev_port? Shouldn't only of the two be appended?

**Comment by keszybz** [comment]
Kernel discussion: http://www.spinics.net/lists/netdev/msg321055.html.

File: src/udev/udev-builtin-net_id.c (hunk 1 above)
Comment (jpirko) [reply src/udev/udev-builtin-net_id.c:176]: There is no in-tree driver implementing both. And it does not really make sense for driver to do that. dev_port will slowly die, phys_port_name is to be used for new drivers.

**Comment by keszybz** [comment]
On Fri, Oct 28, 2016 at 11:47:21AM -0700, Jiří Pírko wrote:

> jpirko commented on this pull request.
> 
> > ```
> >      s = names->pci_onboard;
> >      l = sizeof(names->pci_onboard);
> >      l = strpcpyf(&s, l, "o%d", idx);
> >      if (dev_port > 0)
> >              l = strpcpyf(&s, l, "d%d", dev_port);
> > ```
> > -        if (port_name)
> > -                l = strpcpyf(&s, l, "n%s", port_name);
> 
> There is no in-tree driver implementing both. And it does not really make sense for driver to do that. dev_port will slowly die, phys_port_name is to be used for new drivers.

Hm, and are existing drivers going to switch to phys_port_name? I'd imagine
that in that case they could support both for a time, and phys_port_name
should take precedence?

**Comment by jpirko** [comment]
Sat, Oct 29, 2016 at 12:47:42AM CEST, notifications@github.com wrote:

> On Fri, Oct 28, 2016 at 11:47:21AM -0700, Jiří Pírko wrote:
> 
> > jpirko commented on this pull request.
> > 
> > > ```
> > >      s = names->pci_onboard;
> > >      l = sizeof(names->pci_onboard);
> > >      l = strpcpyf(&s, l, "o%d", idx);
> > >      if (dev_port > 0)
> > >              l = strpcpyf(&s, l, "d%d", dev_port);
> > > ```
> > > -        if (port_name)
> > > -                l = strpcpyf(&s, l, "n%s", port_name);
> > 
> > There is no in-tree driver implementing both. And it does not really make sense for driver to do that. dev_port will slowly die, phys_port_name is to be used for new drivers.
> 
> Hm, and are existing drivers going to switch to phys_port_name? I'd imagine
> that in that case they could support both for a time, and phys_port_name
> should take precedence?

I'm afraid that the current drivers have to stick with dev_port support.
But anyway, this has to work for current, future and past kernels.

**Comment by keszybz** [comment]
Yes, this backward and forward compatibility has to be taken into account.
So why not do:

```
if (port_name)
        l = strpcpyf(&s, l, "n%s", port_name); 
else if (dev_port > 0)
        l = strpcpyf(&s, l, "d%d", dev_port);
```

?

**Comment by jpirko** [comment]
Ok. I can do it like this. I guess I should open up a new pull request right? I'm new to this github thing :( Thanks for review!

**Comment by keszybz** [comment]
No need to open a new one. Just do 'push -f'.

File: src/udev/udev-builtin-net_id.c [hunk 2]
```diff
@@ -35,10 +35,12 @@
...
  *   c<bus_id>                             — CCW bus group name, without leading zeros [s390]
- *   o<index>[d<dev_port>]                 — on-board device index number
- *   s<slot>[f<function>][d<dev_port>]     — hotplug slot index number
+ *   o<index>[d<dev_port>][n<phys_port_name>]
+ *                                         — on-board device index number
+ *   s<slot>[f<function>][d<dev_port>][n<phys_port_name>]
+ *                                         — hotplug slot index number
  *   x<MAC>                                — MAC address
- *   [P<domain>]p<bus>s<slot>[f<function>][d<dev_port>]
+ *   [P<domain>]p<bus>s<slot>[f<function>][d<dev_port>][n<phys_port_name>]
```
Comment (keszybz) [inline src/udev/udev-builtin-net_id.c]: Hm, then this needs changing too:
`[P<domain>]p<bus>s<slot>[f<function>][n<phys_port_name>|d<dev_port>]` ?

Comment (jpirko) [reply src/udev/udev-builtin-net_id.c]: will fix

**Pull request merged**